docker/docker_image_dest.go

```diff
@@ -196,7 +196,7 @@ func (d *dockerImageDestination) HasBlob(info types.BlobInfo) (bool, int64, erro
 		return true, getBlobSize(res), nil
 	case http.StatusUnauthorized:
 		logrus.Debugf("... not authorized")
-		return false, -1, errors.Errorf("not authorized to read from destination repository %s", reference.Path(d.ref.ref))
+		return false, -1, client.HandleErrorResponse(res)
```

Comment:
I like the old and the new message both.  Could we wrap the new and old together?  Something like:

```
errors.Wrapf(client.HandleErrorResponse(res), "\nnot authorized to read from destination repository %s", reference.Path(d.ref.ref))
```
so we'd end up with something like:

```
FATA[0003] Error checking for blob sha256:57310166fe88e0dc63a80ca5c219283a932db0f3969712e2f8a86ada143bf566 at destination: errors:
denied: requested access to the resource is denied
error parsing HTTP 401 response body: unexpected end of JSON input: ""
not authorized to read from destination repository redhat/busybox
```
I like the idea of having the repo that's not reachable be in the error message, but am on the fence about that perhaps being too long.  Yeah, I'm greedy and want both.  What do you all think?  I'm fine with rewording that last line if we can make it fit/feel better, the biggy for me is getting the name out there.

Reply:
The format string passed to `Wrapf` is prepended in front of, not appended to, the wrapped message. So, a
```go
errors.Wrapf(client.HandleErrorResponse(res), "not authorized to read from destination repository %s", reference.Path(d.ref.ref))
```
would end up as
```
FATA[0003] Error checking for blob sha256:57310166fe88e0dc63a80ca5c219283a932db0f3969712e2f8a86ada143bf566 at destination: not authorized to read from destination repository redhat/busybox: errors:
denied: requested access to the resource is denied
error parsing HTTP 401 response body: unexpected end of JSON input: ""
```